ui/src/api/index.ts

```diff
@@ -0,0 +1,16 @@
+import axios from "axios";
+import { API_URL } from "../config/constants";
+import { ICategory } from "../store/category";
+export interface Api {
+	categories(): Promise<ICategory>;
+}
+
+export class Hub implements Api {
+	async categories() {
+		try {
+			return axios.get(`${API_URL}/categories`);
+		} catch (err) {
+			return err.response;
+		}
```

Comment:
Where would be the logic implemented for calling the API after some interval?

Reply:
Once we create the store then we can implement the logic there